### PR TITLE
PHPC-313: BSON should throw when encountering 64-bit integer on 32-bit platform

### PR DESCRIPTION
**Pull request by derickr**

### CODE REVIEW

File: phongo_compat.h [hunk 1]
```diff
@@ -150,23 +150,15 @@
 #elif SIZEOF_PHONGO_LONG == 4
 # define ADD_INDEX_INT64(zval, index, value) \
     if (value > INT32_MAX || value < INT32_MIN) { \
-        char *tmp; \
-        int tmp_len; \
         mongoc_log(MONGOC_LOG_LEVEL_WARNING, MONGOC_LOG_DOMAIN, "Integer overflow detected on your platform: %lld", value); \
-        tmp_len = spprintf(&tmp, 0, "%lld", value); \
-        ADD_INDEX_STRINGL(zval, index, tmp, tmp_len); \
-        efree(tmp); \
+        phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Integer overflow detected on your platform: %lld", value); \
```
Comment (jmikola) [inline phongo_compat.h]: There is little value in continuing to call `mongoc_log()` if we're throwing a proper exception.

File: tests/bson/bug0313.phpt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    '{ "x": { "$numberLong": "4294967294" }}',
+    '{ "x": { "$numberLong": "4294967295" }}',
+    '{ "x": { "$numberLong": "9223372036854775807" }}',
+    '{ "longFieldName": { "$numberLong": "4294967294" }}',
+    '{ "longFieldName": { "$numberLong": "4294967295" }}',
+    '{ "longFieldName": { "$numberLong": "9223372036854775807" }}',
+];
+
+foreach ($tests as $json) {
+	throws(function() use ($json) { var_dump(toPHP(fromJSON($json))); }, 'MongoDB\Driver\Exception\InvalidArgumentException');
```
Comment (jmikola) [inline tests/bson/bug0313.phpt]: This appears to be a tab. PHPT tests should be using 4-space indents.

File: tests/bson/bug0592.phpt [hunk 3]
```diff
@@ -24,14 +24,20 @@ $tests = [
...
+        echo "MongoDB\Driver\Exception\InvalidArgumentException: ", $e->getMessage(), "\n";
+    }
     echo "\n";
 }
 
 ?>
 ===DONE===
 <?php exit(0); ?>
 --EXPECTF--
+%a
```
Comment (jmikola) [inline tests/bson/bug0592.phpt]: Per my earlier comment about `mongoc_log()` being useless, I think we should remove this and the `--INI--` block.

File: phongo_compat.h (hunk 1 above)
Comment (derickr) [inline phongo_compat.h]: OK, will remove it.

File: tests/bson/bug0313.phpt (hunk 2 above)
Comment (derickr) [inline tests/bson/bug0313.phpt]: whatever. I'd rather see .c and .h files following coding standards though. phongo_compat.h is all spaces...

File: tests/bson/bug0592.phpt (hunk 3 above)
Comment (derickr) [inline tests/bson/bug0592.phpt]: Yes.

File: tests/bson/bug0313.phpt (hunk 2 above)
Comment (jmikola) [inline tests/bson/bug0313.phpt]: The integer macros in `phongo_compat.h` is evidently my fault from https://github.com/mongodb/mongo-php-driver/commit/6bcba59cb57a92f091b29e4f11e1c99a96eaed95. I'll clean that up after this and #435 are merged if any offending lines remain.

That's not relevant to the PHPT indentation, though.

**Pull request merged**